api/src/certification/complementary-certification/domain/usecases/attach-badges-to-complementary-certification.js

```diff
@@ -0,0 +1,151 @@
+import { NotFoundError } from '../../../../../lib/domain/errors.js';
+import { DomainTransaction } from '../../../../../lib/infrastructure/DomainTransaction.js';
+import uniq from 'lodash/uniq.js';
+
+import bluebird from 'bluebird';
+import { ComplementaryCertificationBadgeToAttach } from '../models/ComplementaryCertificationBadgeToAttach.js';
+import { InvalidBadgeLevelError } from '../../../shared/domain/errors.js';
+
+const attachBadgesToComplementaryCertification = async function ({
+  complementaryCertificationId,
+  userId,
+  targetProfileIdToDetach,
+  complementaryCertificationBadgesToAttachDTO,
+  ComplementaryCertificationBadgeToAttachValidator,
+  badgeRepository,
+  complementaryCertificationRepository,
+  complementaryCertificationBadgesRepository,
+}) {
+  _verifyThatLevelsAreConsistent({
+    complementaryCertificationBadgesToAttachDTO,
+  });
+
+  await _verifyThatComplementaryCertificationExists({
+    complementaryCertificationId,
+    complementaryCertificationRepository,
+  });
+
+  await _verifyThatBadgesToAttachExist({
+    complementaryCertificationBadgesToAttachDTO,
+    badgeRepository,
+  });
+
+  const complementaryCertificationBadgesToAttach = complementaryCertificationBadgesToAttachDTO.map(
+    (complementaryCertificationBadgeToAttachDTO) => {
+      const complementaryCertificationBadgeToAttach = ComplementaryCertificationBadgeToAttach.from({
+        ...complementaryCertificationBadgeToAttachDTO,
+        complementaryCertificationId,
+        userId,
+      });
+      ComplementaryCertificationBadgeToAttachValidator.validate(complementaryCertificationBadgeToAttach);
+      return complementaryCertificationBadgeToAttach;
+    },
+  );
+
+  return DomainTransaction.execute(async (dtrx) => {
+    const relatedCCBadgesIds = await complementaryCertificationBadgesRepository.getAllIdsByTargetProfileId({
+      targetProfileId: targetProfileIdToDetach,
+    });
+
+    if (relatedCCBadgesIds.length === 0) {
+      throw new NotFoundError('No badges for this target profile.');
+    }
+
+    await _detachExistingComplementaryCertificationBadge({
+      complementaryCertificationBadgesRepository,
+      relatedCCBadgesIds,
+      dtrx,
+    });
+
+    await bluebird.mapSeries(
+      complementaryCertificationBadgesToAttach,
+      async (complementaryCertificationBadgeToAttach) => {
+        await _attachNewComplementaryCertificationBadge({
+          complementaryCertificationBadgesRepository,
+          complementaryCertificationBadge: complementaryCertificationBadgeToAttach,
+          dtrx,
```

Comment:
Remarque : si on peut éviter ça et être explicite

```suggestion
          domainTransaction,
```